pkg/rancher-desktop/utils/string-encode.ts

```diff
@@ -0,0 +1,11 @@
+export const hexEncode = (str: string): string => {
+  let hex = '';
+
+  for (let i = 0; i < str.length; i++) {
+    const charCode = str.charCodeAt(i).toString(16);
+
+    hex += charCode.length < 2 ? `0${ charCode }` : charCode;
```

Comment:
This can also be expressed as
```ts
`0${ charCode }`.slice(-2)
```

Which means we can squash the function to one line (if that looks better…)
```ts
return Array.from(str).map(c => `0${ c.charCodeAt(0).toString(16) }`.slice(-2)).join('');
```

Reply:
Quite succint. I like this implementation better.